<a href="https://colab.research.google.com/github/crpytoscooby/College/blob/master/Riya_K(9063)_ML_1A(Heart).ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

#**Machine Learning Lab Assignment 1 (Group-A)**

**Riya Karambelkar**

**TE IT**

**Roll No 9063**

**Heart**

Importing numpy and pandas libraries

In [ ]:
import numpy as np
import pandas as pd 

Loading the Heart.csv dataset in the data variable

In [ ]:
data = pd.read_csv("/content/Heart.csv")

#Performing operations on the given dataset:

Printing the first five rows

In [ ]:
data.head()

Unnamed: 0   Age  Sex     ChestPain  ...  Slope   Ca        Thal  AHD
0           1  63.0    1       typical  ...    3.0  0.0       fixed   No
1           2  67.0    1  asymptomatic  ...    2.0  3.0      normal  Yes
2           3  67.0    1  asymptomatic  ...    2.0  2.0  reversable  Yes
3           4  37.0    1    nonanginal  ...    3.0  0.0      normal   No
4           5  41.0    0    nontypical  ...    1.0  0.0      normal   No

[5 rows x 15 columns]

Printing the last five rows

In [ ]:
data.tail()

Unnamed: 0   Age  Sex     ChestPain  ...  Slope   Ca        Thal  AHD
298         299  45.0    1       typical  ...    2.0  0.0  reversable  Yes
299         300  68.0    1  asymptomatic  ...    2.0  2.0  reversable  Yes
300         301  57.0    1  asymptomatic  ...    2.0  1.0  reversable  Yes
301         302  57.0    0    nontypical  ...    2.0  1.0      normal  Yes
302         303  38.0    1    nonanginal  ...    1.0  NaN      normal   No

[5 rows x 15 columns]

Printing the shape of data

In [ ]:
data.shape

(303, 15)

This means there are 303 rows and 15 columns in the dataframe

Printing the information of the dataframe

In [ ]:
data.describe()

Unnamed: 0         Age         Sex  ...     Oldpeak       Slope          Ca
count  303.000000  302.000000  303.000000  ...  302.000000  301.000000  298.000000
mean   152.000000   54.476821    0.679868  ...    1.043046    1.598007    0.667785
std     87.612784    9.029545    0.467299  ...    1.161452    0.617411    0.935839
min      1.000000   29.000000    0.000000  ...    0.000000    1.000000    0.000000
25%     76.500000   48.000000    0.000000  ...    0.000000    1.000000    0.000000
50%    152.000000   56.000000    1.000000  ...    0.800000    2.000000    0.000000
75%    227.500000   61.000000    1.000000  ...    1.600000    2.000000    1.000000
max    303.000000   77.000000    1.000000  ...    6.200000    3.000000    3.000000

[8 rows x 12 columns]

In [ ]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 303 entries, 0 to 302
Data columns (total 15 columns):
 #   Column      Non-Null Count  Dtype  
---  ------      --------------  -----  
 0   Unnamed: 0  303 non-null    int64  
 1   Age         302 non-null    float64
 2   Sex         303 non-null    int64  
 3   ChestPain   303 non-null    object 
 4   RestBP      299 non-null    float64
 5   Chol        301 non-null    float64
 6   Fbs         301 non-null    float64
 7   RestECG     301 non-null    float64
 8   MaxHR       302 non-null    float64
 9   ExAng       302 non-null    float64
 10  Oldpeak     302 non-null    float64
 11  Slope       301 non-null    float64
 12  Ca          298 non-null    float64
 13  Thal        301 non-null    object 
 14  AHD         303 non-null    object 
dtypes: float64(10), int64(2), object(3)
memory usage: 35.6+ KB


Finding if there are any missing values in the data

In [ ]:
data.isna()

Unnamed: 0    Age    Sex  ChestPain  ...  Slope     Ca   Thal    AHD
0         False  False  False      False  ...  False  False  False  False
1         False  False  False      False  ...  False  False  False  False
2         False  False  False      False  ...  False  False  False  False
3         False  False  False      False  ...  False  False  False  False
4         False  False  False      False  ...  False  False  False  False
..          ...    ...    ...        ...  ...    ...    ...    ...    ...
298       False  False  False      False  ...  False  False  False  False
299       False  False  False      False  ...  False  False  False  False
300       False  False  False      False  ...  False  False  False  False
301       False  False  False      False  ...  False  False  False  False
302       False  False  False      False  ...  False   True  False  False

[303 rows x 15 columns]

In [ ]:
print("Total missing values in the dataframe are: ")
print(data.isna().sum())

Total missing values in the dataframe are: 
Unnamed: 0    0
Age           1
Sex           0
ChestPain     0
RestBP        4
Chol          2
Fbs           2
RestECG       2
MaxHR         1
ExAng         1
Oldpeak       1
Slope         2
Ca            5
Thal          2
AHD           0
dtype: int64


Finding data type of each column 

In [ ]:
data.dtypes

Unnamed: 0      int64
Age           float64
Sex             int64
ChestPain      object
RestBP        float64
Chol          float64
Fbs           float64
RestECG       float64
MaxHR         float64
ExAng         float64
Oldpeak       float64
Slope         float64
Ca            float64
Thal           object
AHD            object
dtype: object

Finding out zeros in the dataframe

In [ ]:
data.all()

Unnamed: 0     True
Age            True
Sex           False
ChestPain      True
RestBP         True
Chol           True
Fbs           False
RestECG       False
MaxHR          True
ExAng         False
Oldpeak       False
Slope          True
Ca            False
Thal           True
AHD            True
dtype: bool

Hence the columns Sex, Fbs, RestECG, ExAng, Oldpeak and Ca contain zeros.

Counting the number of zeros in each column

In [ ]:
(data==0).sum()

Unnamed: 0      0
Age             0
Sex            97
ChestPain       0
RestBP          0
Chol            0
Fbs           257
RestECG       149
MaxHR           0
ExAng         204
Oldpeak        98
Slope           0
Ca            176
Thal            0
AHD             0
dtype: int64

Finding the mean age of patients

In [ ]:
data['Age'].mean()

54.47682119205298

Finding the maximum age in the dataframe

In [ ]:
data['Age'].max()

77.0

Replacing missing values in the Age column by the mean age

In [ ]:
data['Age'].fillna(data['Age'].mean())

0      63.0
1      67.0
2      67.0
3      37.0
4      41.0
       ... 
298    45.0
299    68.0
300    57.0
301    57.0
302    38.0
Name: Age, Length: 303, dtype: float64

Extracting only Age, Sex, ChestPain, RestBP and Chol columns from the dataset

In [ ]:
data1=data[['Age', 'Sex', 'ChestPain', 'RestBP', 'Chol']]
data1

Age  Sex     ChestPain  RestBP   Chol
0    63.0    1       typical   145.0  233.0
1    67.0    1  asymptomatic   160.0  286.0
2    67.0    1  asymptomatic   120.0  229.0
3    37.0    1    nonanginal   130.0  250.0
4    41.0    0    nontypical   130.0  204.0
..    ...  ...           ...     ...    ...
298  45.0    1       typical   110.0    NaN
299  68.0    1  asymptomatic   144.0  193.0
300  57.0    1  asymptomatic   130.0  131.0
301  57.0    0    nontypical   130.0  236.0
302  38.0    1    nonanginal     NaN  175.0

[303 rows x 5 columns]

Replacing the missing values with mean values

In [ ]:
data1['RestBP'].mean()

131.57190635451505

In [ ]:
data1['Chol'].mean()

246.50498338870432

In [ ]:
data1 = data1.fillna(data1.mean())
data1

Age  Sex     ChestPain      RestBP        Chol
0    63.0    1       typical  145.000000  233.000000
1    67.0    1  asymptomatic  160.000000  286.000000
2    67.0    1  asymptomatic  120.000000  229.000000
3    37.0    1    nonanginal  130.000000  250.000000
4    41.0    0    nontypical  130.000000  204.000000
..    ...  ...           ...         ...         ...
298  45.0    1       typical  110.000000  246.504983
299  68.0    1  asymptomatic  144.000000  193.000000
300  57.0    1  asymptomatic  130.000000  131.000000
301  57.0    0    nontypical  130.000000  236.000000
302  38.0    1    nonanginal  131.571906  175.000000

[303 rows x 5 columns]

Dividing the dataset in training(75%) and testing(25%)

In [ ]:
from sklearn.model_selection import train_test_split

train_data, test_data = train_test_split(data1, test_size=0.25)

In [ ]:
train_data.shape

(227, 5)

In [ ]:
test_data.shape

(76, 5)